### PR TITLE
Feature/tel 394

### PR DESCRIPTION
**Pull request by oyvindbergbouvet**
New step indicator: 
https://www.figma.com/file/qBaEEcuzYMlHOysEHmzyKJ/FWA---Tr%C3%A5dl%C3%B8st-Bredb%C3%A5nd?node-id=827%3A7951

It has a few options, and it supports a couple of scenarios instead of "just a step indicator, not clickable, just reading current url as a step indicator"... which we already have.

This is not updating the existing one - this contains "paging" of the steps with a lot of options and supports various ways of "moving" (click inside a component, click on the step indicator, navigate to new url [full page refresh], or disable all clicks)...
- also: so teams can adjust to it from the old "step indicator" whenever they got time, if they need/want

### CODE REVIEW

File: src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx [hunk 1]
```diff
@@ -65,7 +65,7 @@ export const MinimumRightHeading = () => {
   );
 };
 
-export const SimpleNoIconNoBadgeNoIngress = () => {
+export const TitleLeftAndRightOnly = () => {
```
Comment (oyvindbergbouvet) [inline src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx:68]: Just reworded the examples in the "Accordion Flexible" comp

File: src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx [hunk 2]
```diff
@@ -124,10 +124,23 @@ export const NoIconWithBadge = () => {
   );
 };
 
-export const MultipleIconsNoBadge = () => {
+export const TitleLeftAndLineThroughRight = () => {
```
Comment (oyvindbergbouvet) [inline src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx:127]: And added one more story

File: src/colors.pcss [hunk 3]
```diff
@@ -111,6 +111,7 @@ Additional colors
   --alert-blue: #e5f4ff;
   --alert-dark-blue: #0099ff;
   --blue: #00cdff;
+  --darker-grey: #606064;
```
Comment (oyvindbergbouvet) [inline src/colors.pcss:114]: Used as border for "step indicator buttons" (circles)

File: src/molecules/AccordionFlexible/AccordionFlexible.pcss [hunk 4]
```diff
@@ -139,6 +139,7 @@
       flex-basis: auto;
       flex-shrink: 1;
       flex-grow: 1;
+      align-items: start;
```
Comment (oyvindbergbouvet) [inline src/molecules/AccordionFlexible/AccordionFlexible.pcss:142]: A minor issue I spotted - so just "did it"

File: src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx [hunk 5]
```diff
@@ -65,7 +65,7 @@ export const MinimumRightHeading = () => {
   );
 };
 
-export const SimpleNoIconNoBadgeNoIngress = () => {
```
Comment (oyvindbergbouvet) [inline src/molecules/AccordionFlexible/AccordionFlexible.stories.tsx:68]: Just reworded a few stories in a different "component" (molecule)

File: src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,103 @@
+import React from 'react';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsComponents.tsx:1]: A "data object" (array/list) where children are HTML for testing various stories

File: src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsHtml.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
+import { Step } from '../Models/Step';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsHtml.tsx:1]: A "data object" (array/list) where children are pure html STRINGS for testing various stories

File: src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsHtmlWithUrls.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
+import { Step } from '../Models/Step';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Data/StepDataChildrenAsHtmlWithUrls.tsx:1]: A "data object" (array/list) where children are pure html strings for testing various stories, where URL are defined so clicking "steps" will hopefully navigate to the url specified on the step

File: src/molecules/StepIndicatorPaging/Functions/getMinStepNumber.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+  if (!forward) {
+    if (number > max) {
+      number = max;
+    } else {
+      number = number - 1;
+    }
+  }
+
+  return number;
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Functions/getMinStepNumber.tsx:25]: Returns the minimun step number to be displayed based on the input, which step are we at? and what is the "pageSize" (max amount of steps to show)
- could rename "pageSize" to "stepSize"... but "page" and "paging logic" is common for components like: paging, carrousel, and "step indicators" (that actually are not just simple indicators, but clickable...)

File: src/molecules/StepIndicatorPaging/Functions/navigateToStepUrl.tsx [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export const navigateToStepUrl = (url: any) => {
+  if (url && url.length > 0) {
+    window.location.assign(url);
+    return true;
+  }
+};
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Functions/navigateToStepUrl.tsx:6]: Whenever a step is clicked, a function to check if url on step is defined to we just "go to"

File: src/molecules/StepIndicatorPaging/Functions/onPagingLeft.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import { navigateToStepUrl } from './navigateToStepUrl';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Functions/onPagingLeft.tsx:1]: Abstracted pagingLeft and pagingRight logic for clarity

File: src/molecules/StepIndicatorPaging/Functions/setStepsComplete.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import { Step } from '../Models/Step';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Functions/setStepsComplete.tsx:1]: Abstracted SetStepsCompleted for clarity

File: src/molecules/StepIndicatorPaging/Models/InternalStep.ts [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import { Step } from './Step';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/Models/InternalStep.ts:1]: Models/ contains the interfaces used throughout the molecule

File: src/molecules/StepIndicatorPaging/RenderArrow.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/RenderArrow.tsx:1]: Rendering the "arorw buttons", both left and right arrows, at each side of the step indicator

File: src/molecules/StepIndicatorPaging/StepIndicatorPaging.pcss [hunk 15]
```diff
@@ -0,0 +1,411 @@
...
+    padding: 0;
+    padding-bottom: 24px;
+    width: min-content;
+    position: relative;
+    margin: 0 auto;
+
+    @media screen and (min-width: 48em) {
+      padding-bottom: 40px;
+    }
+    /*TODO/Optional: Update to display flex - grow, with 'min and max value per list item....' ? */
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/StepIndicatorPaging.pcss:16]: The todo, could remove it or do it... but time, want this first release out, so we can try to use it in telia.no and "feel it"

File: src/molecules/StepIndicatorPaging/StepIndicatorPaging.pcss [hunk 16]
```diff
@@ -0,0 +1,411 @@
...
+        & > svg {
+          display: none;
+        }
+        & > span {
+          display: block;
+        }
+      }
+    }
+
+    &-clickable-field {
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/StepIndicatorPaging.pcss:148]: Just making the "clickable area" slightly larger than buttons are

File: src/molecules/StepIndicatorPaging/RenderStepButton.tsx [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+
+  return (
+    <>
+      {!isClickable && (
+        <span className={buttonClassNames}>
+          <RenberButtonText />
+        </span>
+      )}
+      {isClickable && (
+        <button role="button" onClick={onStepClick} className={buttonClassNames} tabIndex={!isActive ? number + 1 : -1}>
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/RenderStepButton.tsx:51]: TabIndex is -1 not tab-able if we are on this step...

File: src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx [hunk 18]
```diff
@@ -0,0 +1,297 @@
...
+import { getMinStepNumberInRange } from './Functions/getMinStepNumberInRange';
+import { onPagingLeft } from './Functions/onPagingLeft';
+import { onPagingRight } from './Functions/onPagingRight';
+import { setStepsComplete } from './Functions/setStepsComplete';
+import { RenderArrow } from './RenderArrow';
+import classNames from 'classnames';
+
+// eslint-disable-next-line react/display-name
+const StepIndicatorPaging = React.forwardRef((props: Props, ref) => {
+  useImperativeHandle(ref, () => ({
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx:18]: Supporting clickable "onStepButtons" in the components children
- if one specify the "buttonId's" when creating the step indicator

File: src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx [hunk 19]
```diff
@@ -0,0 +1,297 @@
...
+  } = props;
+
+  const completePreviousSteps = navigationCompletesPreviousSteps != false;
+  const pageSize = props.pageSize ?? 5;
+  const maxStepCount = props.steps.length - 1;
+  const initialStepNumber =
+    props.initialStepNumber != null && props.initialStepNumber > 0 ? props.initialStepNumber - 1 : 0;
+
+  if (pageSize < 3 || pageSize > 5) {
+    console.error('StepIndicatorPaging: PageSize is outside of tested range');
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx:53]: Currently only supporting 3-5 in pagesize, should be fine.

File: src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx [hunk 20]
```diff
@@ -0,0 +1,297 @@
...
+    currentActiveStepNumber: initialStepNumber,
+  });
+
+  const internalSteps: InternalStep[] = state.steps.map((step, i) => {
+    return {
+      number: i,
+      title: step.title,
+      url: step.url,
+      isComplete: step.isComplete || (completePreviousSteps && i < state.currentActiveStepNumber),
+      arrowType: null,
```
Comment (oyvindbergbouvet) [inline src/molecules/StepIndicatorPaging/StepIndicatorPaging.tsx:72]: simply "converting" (mapping) our Step object to "InternalStep" object
- which includes arrowType param which we will set later if arrows should show...

**Review by remibengtbouvet — APPROVED** [review]

**Pull request merged**